packages/mint-components/src/components/sqm-leaderboard/sqm-leaderboard-view.tsx

```diff
@@ -18,41 +22,156 @@ export interface LeaderboardViewProps {
     loadingstate: VNode;
   };
 }
 
+const style = {
+  Leaderboard: {
+    "& table": {
+      width: "100%",
+      borderCollapse: "collapse",
+    },
+    "& th": {
+      textAlign: "left",
+      padding: "var(--sl-spacing-medium)",
+      paddingTop: "0",
+      fontSize: "var(--sl-font-size-medium)",
+      fontWeight: "var(--sl-font-weight-semibold)",
+    },
+    "& tr:not(:first-child)": {
+      borderTop: "1px solid var(--sl-color-neutral-200)",
+    },
+    "& td": {
+      fontSize: "var(--sl-font-size-medium)",
+      fontWeight: "var(--sl-font-weight-normal)",
+    },
+    "& .ellipses": {
+      textAlign: "center",
+      padding: "0",
+      color: "var(--sl-color-neutral-500)",
+    },
+    "& .highlight": {
+      background: "var(--sl-color-primary-50)",
+    },
+    "& td, th": {
+      color: "var(--sl-color-gray-800)",
+      padding: "var(--sl-spacing-medium)",
+    },
+    "& .User": {
+      width: "100%",
+    },
+    "& .Score": {
+      width: "auto",
+      whiteSpace: "nowrap",
+    },
+  },
+};
+
+const sheet = createStyleSheet(style);
+const styleString = sheet.toString();
+
+const vanillaStyle = `
+	:host{
+		display: block;
+	}
+`;
+
 export function LeaderboardView(props: LeaderboardViewProps) {
   const { states, data, elements } = props;
   const { styles } = states;
-  if (states.loading) {
-    return elements.loadingstate;
-  }
+
+  if (states.loading)
+    return (
+      <div class={sheet.classes.Leaderboard}>
+        <style type="text/css">
+          {styleString}
+          {vanillaStyle}
+        </style>
+        {elements.loadingstate}
+      </div>
+    );
+
+  if (!states.hasLeaders) return elements.empty;
+
+  let userSeenFlag = false;
+
   return (
-    <div>
-      {!states.hasLeaders && elements.empty}
-      {states.hasLeaders && (
-        <table>
+    <div class={sheet.classes.Leaderboard}>
+      <style type="text/css">
+        {styleString}
+        {vanillaStyle}
+      </style>
+      <table>
+        <tr>
+          {styles.showRank && <th class="Rank">{styles.rankheading}</th>}
+          <th class="User">{styles.usersheading}</th>
+          <th class="Score">{styles.statsheading}</th>
+        </tr>
+        {data.leaderboard?.map((user) => {
+          if (user.rowNumber === data.viewerRank?.rowNumber)
+            userSeenFlag = true;
+          return (
+            <tr
+              class={
+                !styles.hideViewer &&
+                user.rowNumber === data.viewerRank?.rowNumber
+                  ? "highlight"
+                  : ""
+              }
+            >
+              {styles.showRank && <td class="Rank">{user.rank}</td>}
+              <td class="User">
+                {user.firstName && user.lastInitial
+                  ? user.firstName + " " + user.lastInitial
+                  : user.firstName || user.lastInitial
+                  ? user.firstName || user.lastInitial
+                  : styles.anonymousUser || "Anonymous User"}
```

Comment:
If the prop has a default value, then why are we using a hardcoded fallback in the code?

Reply:
removed!